server/aesdsocket.c

```diff
@@ -0,0 +1,454 @@
+ * @return -1 otherwise
+ */
+static int read_line_from_stream(FILE *stream, char **line, size_t *line_len);
+
+/**
+ * @brief Sends the `line` to the `client_fd`
+ * @param client_fd client socket
+ * @param line string to send to the client
+ * @return 0 if successful
+ * @return -1 otherwise
+ */
+static int send_line(const int client_fd, char *line);
+
+/**
+ * @brief Sends the contents of `file` to the `client_fd` one line at a time
+ * @param file file to send
+ * @param client_fd client socket
+ * @return 0 if successful
+ * @return -1 otherwise
+ */
+static int send_file(char *file, const int client_fd);
+
+/**
+ * @brief SIGINT signal handler used to gracefully shutdown the program
+ * @param sig signal number
+ */
+void sigint_handler(int sig);
+
+/**
+ * @brief Runs the socket application
+ * @param server_info address information for the server
+ * @param is_daemon indicates if application should be run as a daemon
+ * @return 0 if successful
+ * @return -1 otherwise
+ */
+int application(struct addrinfo **server_info, const bool is_daemon);
+
+int main(int argc, char *argv[]) {
+  openlog("aesdsocket", LOG_PID, LOG_USER);
+
+  bool execute_as_daemon = false;
+  if ((argc == 2) && (strcmp(argv[1], "-d") == 0)) {
+    execute_as_daemon = true;
+  } else if (argc != 1) {
+    printf("Usage: %s [-d]\r\n", argv[0]);
+    printf("Options:\r\n");
+    printf("  -d    execute as deamon\r\n");
+    closelog();
+    return 1;
+  }
+
+  // Register SIGINT handler
+  if (signal(SIGINT, sigint_handler) == SIG_ERR) {
+    perror("signal");
+    closelog();
+    return -1;
+  }
+
+  // Register SIGTERM handler
+  if (signal(SIGTERM, sigint_handler) == SIG_ERR) {
+    perror("signal");
+    closelog();
+    return -1;
+  }
+
+  // Run application
+  struct addrinfo *server_addrinfo = NULL;
+  int result = application(&server_addrinfo, execute_as_daemon);
+
+  // Clean up
+  freeaddrinfo(server_addrinfo);
+  closelog();
+  return result;
+}
+
+int application(struct addrinfo **server_info, const bool is_daemon) {
+  syslog(LOG_DEBUG, "Starting `aesdsocket`.");
+
+  // Setup the server
+  int server_socket = 0;
+  if (create_server_socket(&server_socket, server_info) != 0) {
+    syslog(LOG_ERR, "create_server_socket");
+    close(server_socket);
+    return -1;
+  }
+
+  if (is_daemon) {
+    const int daemon_resut = daemonize();
+    switch (daemon_resut) {
+    case -1:
+      syslog(LOG_ERR, "daemonize");
+      close(server_socket);
+      return -1;
+    case 1:
+      exit(EXIT_SUCCESS);
+    default:
+    }
+  }
+
+  // Loop until termination signal is received
+  while (!is_terminated) {
+    // Wait for a connection
+    int client_socket = 0;
+    const int connection_result =
+        create_client_connection(server_socket, &client_socket);
+    switch (connection_result) {
+    case -1: // error
+      syslog(LOG_ERR, "create_client_connection");
+      close(client_socket);
+      close(server_socket);
+      return -1;
```

Comment:
I think you are missing cleanup of `RESULT_FILE` in a few of these code paths within the `while` loop.